### DISCUSSION TITLE
Store private Key and sign, verify in Nada

### DISCUSSION THREAD

**Discussion by vinitran** [NillionNetwork/discussions#45] [Q&A] [answered]
As a title, how can i deal with it ? i want to sign and verify in Nada program

**Comment by oceans404** [comment, marked as answer]
Hey @vinitran, great question! 

Today you can store a private key in Nillion as a SecretBlob

- Python Client - [Store SecretBlob](https://docs.nillion.com/store-secrets#store-a-secretblob)
- JavaScript Client - [Store SecretBlob](https://docs.nillion.com/store-secrets-js#store-a-secretblob)

You can't yet do an ECSDSA signature (sign and verify with Nada) but we're working on supporting this use case.

**Reply by Davetbutler** [reply]
Hey @vinitran 

Dave from the product team here.

We are starting to spec out our signatures feature. Would you mind helping me understand your use case a bit more?

**Comment by MusaDauda** [comment]
That's good to see. Keep it up, guys! 👏

**Reply by Davetbutler** [reply]
Hey @MusaDauda. 

Dave from the product team here. I see you are also interested in using signatures on Nillion.

We are starting to spec out our signatures feature. Would you mind helping me understand your use case a bit more?

**Comment by phuongvietvu0306** [comment]
@oceans404 is the ECSDSA signature supported yet? If not, how long the way can Nillion support this feature?

**Reply by oceans404** [reply]
Signatures aren't supported yet, but we plan to support them soon

**Reply by Davetbutler** [reply]
Hey @phuongvietvu0306. Dave from the product team here.

We are starting to spec out our signatures feature. Would you mind helping me understand your use case a bit more?
